packages/blob/src/client.ts

```diff
@@ -0,0 +1,72 @@
+import type { BodyInit } from 'undici';
+// When bundled via a bundler supporting the `browser` field, then
+// the `undici` module will be replaced with https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API
+// for browser contexts. See ./undici-browser.js and ./package.json
+import { fetch } from 'undici';
+import {
+  BlobAccessError,
+  BlobError,
+  BlobUnknownError,
+  mapBlobResult,
+  type BlobResult,
+  type BlobMetadataApi,
+  type PutCommandOptions,
+  getApiUrl,
+} from '.';
+
+export { type BlobResult } from '.';
+
+export interface ClientPutCommandOptions extends PutCommandOptions {
+  clientToken: string;
+  onUploadCompletedUrl?: string;
+}
+
+export async function put(
+  pathname: string,
+  body: string | Blob | ArrayBuffer | FormData | ReadableStream | File,
+  options: ClientPutCommandOptions,
+): Promise<BlobResult> {
+  if (typeof window === 'undefined') {
+    throw new BlobError('access must be "public"');
+  }
+  if (!pathname) {
+    throw new BlobError('pathname is required');
+  }
+
+  if (!body) {
+    throw new BlobError('body is required');
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  if (!options || options.access !== 'public') {
+    throw new BlobError('access must be "public"');
+  }
+
+  if (!options.clientToken) {
```

Comment:
Can we keep "token"? Since we're already importing from client I do not think we need to change the option name here

Reply:
throwing in case the token doesn't start with `vercel_blob_client`